aten/src/ATen/native/transformers/cuda/sdp_utils.cpp

```diff
@@ -289,6 +289,18 @@ bool check_mem_efficient_hardware_support(sdp_params const& params, bool debug)
   return true;
 }
 
+bool check_mem_efficient_batch_size(sdp_params const& params, bool debug) {
```

Comment:
Doesn't mem efficient attention live in the PyTorch cudabase + xformers? Should we just fix this properly by changing a dtype / type somewhere?

Reply:
mem-efficient lives entirely within PyTorch code base. I think the problem isn't dtype related rather that cuda imposes limits on a singular grid dim size. 

The proper fix would likely be to split this large grid dim among another

Reply:
Or just launching two kernels, right? It's separable along batch.

Reply:
yeah or this